src/data/real/irrational.lean

```diff
@@ -257,4 +257,46 @@ open irrational
 @[simp] theorem irrational_inv_iff : irrational x⁻¹ ↔ irrational x :=
 ⟨of_inv, irrational.inv⟩
 
+/-!
+### Lemmas about Irrationals between Reals
+-/
+
+/-- There exists an irrational number between any two distinct rational numbers.-/
+theorem exists_irrational_btwn_rats {x y :ℚ} (h : (x :ℝ) < y) :
+∃ (r : ℝ), irrational r ∧ (x : ℝ) < r ∧ r < y:=
+begin
+  use x + ((y - x) : ℚ) * (real.sqrt 2)⁻¹,
```

Comment:
`x` and `y` are already in ℚ, so I'm surprised this annotation is needed

Reply:
The notation is needed to prevent lean casting them to `ℝ` before doing the subtraction